src/Analyzers/CSharp/Analyzers/UseInterpolatedString/CSharpUseInterpolatedStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.UseInterpolatedString;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageService;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageService;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseInterpolatedString;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal class CSharpUseInterpolatedStringDiagnosticAnalyzer() : AbstractUseInterpolatedStringDiagnosticAnalyzer<SyntaxKind, ExpressionSyntax, LiteralExpressionSyntax>
+{
+    protected override ISyntaxFacts GetSyntaxFacts() => CSharpSyntaxFacts.Instance;
+
+    protected sealed override bool CanConvertToInterpolatedString(LiteralExpressionSyntax stringLiteralExpression, ParseOptions parseOptions)
+    {
+        // Check if the token is a string literal
+        if (stringLiteralExpression.Kind() != SyntaxKind.StringLiteralExpression)
+            return false;
+
+        // Check the string literal for errors
+        if (stringLiteralExpression.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
+            return false;
+
+        // If there is a const keyword, do not offer the refactoring
+        // An interpolated string is not const
```

Comment:
"An interpolated string is not const" - that is not true. C# 10 added [constant interpolated strings](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/constant_interpolated_strings). In short, if all interpolation variables are constants, the interpolation is considered a constant as well, e.g.:
```cs
const string a = "a";
const string b = $"Value of 'a' is: {a}";
```
Whether to support this case is up to you. If this requires just several touches, then go ahead. If not, feel free to leave it as is. But in either way the comment should be either removed or changed to say that supporting constant interpolations would require too much effort or something similar